packages/flutter/lib/src/widgets/localized_resources.dart

```diff
@@ -0,0 +1,354 @@
+/// classes that contain `Intl.message()` calls. In this case that would just be the
+/// `MyLocalizedResources` class.
+///
+/// One could choose another approach for loading localized resources and looking them up while
+/// still conforming to the structure of this example.
+class LocalizedResources extends StatefulWidget {
+  LocalizedResources({
+    Key key,
+    @required this.locale,
+    this.delegate,
+    this.child
+  }) : super(key: key) {
+    assert(locale != null);
+  }
+
+  /// The resources returned by [LocalizedResources.of] will be specific to this locale.
+  final Locale locale;
+
+  /// This delegate is responsible for loading and caching resources for `locale`.
+  ///
+  /// Typically defined with [DefaultLocalizedResourcesDelegate].
+  final LocalizedResourcesDelegate delegate;
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  /// Returns the localized resources for the widget tree that corresponds to
+  /// [BuildContext] `context`.
+  ///
+  /// This method is typically combined with [LocalizedResourcesData.for] to defined
+  /// a static method that looks up a collection of related related resources. For
+  /// example Flutter looks up Material resources with a method defined like this:
+  ///
+  /// ```dart
+  /// static LocalizedMaterialResources of(BuildContext context) {
+  ///  return LocalizedResources.of(context)
+  ///    .resourcesFor<LocalizedMaterialResources>(LocalizedMaterialResources);
+  /// }
+  /// ```
+  static LocalizedResourcesData of(BuildContext context) {
+    final _LocalizedResourcesScope scope = context.inheritFromWidgetOfExactType(_LocalizedResourcesScope);
+    return scope?.localizedResourcesState;
+  }
+
+  @override
+  _LocalizedResourcesState createState() => new _LocalizedResourcesState();
+}
+
+class _LocalizedResourcesState extends State<LocalizedResources> implements LocalizedResourcesData {
+  final GlobalKey _localizedResourcesScopeKey = new GlobalKey();
+
+  @override
+  Locale get locale => _locale;
+  Locale _locale;
+
+  @override
+  void initState() {
+    super.initState();
+    load(widget.locale);
+  }
+
+  void load(Locale locale) {
+    if (widget.delegate == null) {
+      setState(() {
+        _locale = locale;
+      });
+      return;
+    }
+
+    widget.delegate.load(locale).then((_) {
+      setState(() {
+        _locale = locale;
+      });
+      final InheritedElement scopeElement = _localizedResourcesScopeKey.currentContext;
+      scopeElement?.dispatchDidChangeDependencies();
+    });
+  }
+
+  @override
+  T resourcesFor<T>(Type type) => widget.delegate.resourcesFor<T>(_locale, type);
+
+  @override
+  Widget build(BuildContext context) {
+    return new _LocalizedResourcesScope(
+      key: _localizedResourcesScopeKey,
+      locale: widget.locale,
+      localizedResourcesState: this,
+      child: _locale != null ? widget.child : new Container(),
```

Comment:
I don't think one blank frame is good enough for the level of quality we are going for. If we are talking about initialization during start-up, I think what @xter is working on for iOS right now is very relevant. He just showed me how we can have a splash screen wait until the first useful frame (i.e. one rendered from `runApp`). So if we can structure the initialization API such that we do not call `runApp` until i18n resources (synchronous or not) are ready we shouldn't have blank frames. Of course, similar work would have to be repeated for Android, but we can validate the approach on iOS first.